android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -169,65 +169,63 @@ protected void performUpdate() {
-          // if the previous top screen does not exist anymore and the new top was not on the stack before,
-          // probably replace was called, so we check the animation
-          transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
-        }
-        switch (newTop.getScreen().getStackAnimation()) {
-          case NONE:
-            transition = FragmentTransaction.TRANSIT_NONE;
-            break;
-          case FADE:
-            transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
-            break;
-          case SLIDE_FROM_RIGHT:
-            customAnimation = true;
-            getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_right, R.anim.rns_slide_out_to_left);
-            break;
-          case SLIDE_FROM_LEFT:
-            customAnimation = true;
-            getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
-            break;
-        }
-
-        if (!customAnimation) {
-          getOrCreateTransaction().setTransition(transition);
-        }
+        // if the previous top screen does not exist anymore and the new top was not on the stack before,
+        // probably replace or reset was called, so we play the "back animation"
+        // otherwise it's open animation
+        shouldPlayOpenAnimation = mScreenFragments.contains(mTopScreen) || newTop.getScreen().getReplaceAnimation() != Screen.ReplaceAnimation.POP;
+        // and check if it is a custom animation
+        isCustomAnimation = isCustomAnimation(newTop);
+        stackAnimation = newTop.getScreen().getStackAnimation();
       }
     } else if (mTopScreen != null && !mTopScreen.equals(newTop)) {
       // otherwise if we are performing top screen change we do "back animation"
-      int transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
-
-      switch (mTopScreen.getScreen().getStackAnimation()) {
-        case NONE:
-          transition = FragmentTransaction.TRANSIT_NONE;
-          break;
-        case FADE:
-          transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
-          break;
-        case SLIDE_FROM_RIGHT:
-          customAnimation = true;
-          getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
-          break;
-        case SLIDE_FROM_LEFT:
-          customAnimation = true;
-          getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_right, R.anim.rns_slide_out_to_left);
-          break;
+      shouldPlayOpenAnimation = false;
+      // and check if it is a custom animation
+      isCustomAnimation = isCustomAnimation(mTopScreen);
+      stackAnimation = mTopScreen.getScreen().getStackAnimation();
+    }
+
+    // animation logic start
+    if (shouldPlayOpenAnimation) {
+      transition = FragmentTransaction.TRANSIT_FRAGMENT_OPEN;
+
+      if (stackAnimation == Screen.StackAnimation.SLIDE_FROM_RIGHT) {
```

Comment:
I'd leave those as a `switch` since we are planning to add more transitions.

Reply:
Fixed in 7e8721b08db4ceda2942dc73f00470caff78edd8